src/lib/phdi-building-blocks/phdi_building_blocks/utils.py

```diff
@@ -1,10 +1,175 @@
+    # proper order. All of this is to say that a series of if statements appears to be
+    # the most straightforward means by which we can ensure that the transformations are
+    # applied in the appropriate order.
+    text = raw_text
+
+    if "remove_numbers" in kwargs:
+        text = "".join([ltr for ltr in text if not ltr.isnumeric()])
+    if "remove_punctuation" in kwargs:
+        text = "".join([ltr for ltr in text if ltr.isalnum() or ltr == " "])
+    if "remove_characters" in kwargs:
+        text = "".join([ltr for ltr in text if ltr not in kwargs["remove_characters"]])
+    if "trim" in kwargs:
+        text = text.strip()
+    if "case" in kwargs:
+        if kwargs["case"] == "upper":
+            text = text.upper()
+        if kwargs["case"] == "lower":
+            text = text.lower()
+        if kwargs["case"] == "title":
+            text = text.title()
+
+    return text
+
+
+def http_request_with_retry(
+    url: str,
+    retry_count: int,
+    request_type: str,
+    allowed_methods: List[str],
+    headers: dict,
+    data: dict = None,
+) -> Union[None, requests.Response]:
+    """
+    Carryout an HTTP Request using a specific retry strategy. Essentially
+    a wrapper function around the retry strategy implementation of a
+    mounted HTTP request.
+    :param url: The url at which to make the HTTP request
+    :param retry_count: The number of times to re-try the request, if the
+    first attempt fails
+    :param request_type: The type of request to be made. Currently supports
+    GET and POST.
+    :param allowed_methods: The list of allowed HTTP request methods (i.e.
+    POST, PUT, etc.) for the specific URL and query
+    :param headers: JSON-type dictionary of headers to make the request with,
+    including Authorization and content-type
+    :param data: JSON data in the case that the request requires data to be
+    posted. Defaults to none.
+    """
+    # Configure the settings of the 'requests' session we'll make
+    # the API call with
+    retry_strategy = Retry(
+        total=retry_count,
+        status_forcelist=[429, 500, 502, 503, 504],
+        allowed_methods=allowed_methods,
+    )
+    adapter = HTTPAdapter(max_retries=retry_strategy)
+    http = requests.Session()
+    http.mount("https://", adapter)
+
+    # Now, actually try to complete the API request
+    if request_type == "POST":
+        try:
+            requests.post(
+                url,
+                headers=headers,
+                data=json.dumps(data),
+            )
+        except Exception:
+            logging.exception(
+                "POST request to " + url + " failed for data: " + str(data)
+            )
+            return
+    elif request_type == "GET":
+        try:
+            response = requests.get(
+                url,
+                headers=headers,
+            )
+            return response
+        except Exception:
+            logging.exception("GET request to " + url + " failed.")
```

Comment:
I know this will be one more conversation that has to be resolved, but I just want to say that I think this work here is awesome!